Bio/SeqFeature.py

```diff
@@ -1587,11 +1601,11 @@ def extract(self, parent_sequence, references=None):
         return f_seq
 
 
-class AbstractPosition:
```

Comment:
Removing the ``AbstractPosition`` class would be a backward compatibility break. Rather than replacing it with ``Position(ABC)``, can you not make this into the ABC?

Reply:
Then we get `AbstractPosition.fromstring(...)` which I thought sounded weird. I don't think any user would use 'AbstractPosition`; at least I cannot think of any realistic use case.

Reply:
I agree existing direct usage of ``AbstractPosition`` would be rare. Perhaps with ``isinstance``... which would be frustrating as if we keep AbstractPosition as a deprecated class in the hierarchy with a warning on ``__init__`` that warning would not trigger on with ``isinstance`` would it?

Reply:
That is correct; `isinstance` just checks the class hierarchy; it doesn't create any instances.

Reply:
I'm not sure many people will use ``(Abstract)Position.fromstring(...)`` under either name, so perhaps keeping ``AbstractPosition`` is fine?

The suggested ``FeatureLocation`` method will be more popular, ``FeatureLocation.fromINSDC(...)`` or whatever it gets called.

Reply:
I am thinking

- `Position` as the abstract base class, with subclasses
  - `ExactPosition`
  - `UncertainPosition`
  - `UnknownPosition`
  - `WithinPosition`
  - `BetweenPosition`
  - `BeforePosition`
  - `AfterPosition`
  - `OneOfPosition` 
- `Location` as the abstract base class, with subclasses
  - `SimpleLocation`
  - `CompoundLocation`

and a `fromstring` factory method both on `Position` and on `Location`.

Reply:
Sounds good.

I'm not 100% sold on the name ``fromstring`` since there are many different location string conventions, thus my suggestion ``fromINSDC`` as a suggestion covering the shared convention used in GenBank/EMBL/DDBJ. However, I'm willing to be outvoted, ``fromstring`` is concise.

Reply:
Although I lean towards keeping ``FeatureLocation`` rather than renaming it to ``SimpleLocation`` to reduce code churn. That would require a lot of third party code to be updated.

Reply:
I do think it is important to have clear names for classes. To me, `FeatureLocation` sounds like a feature with a location, while in reality it is the location of a feature. But then is a `CompoundLocation` the location of a compound? An `AbstractPosition` the position of an abstract? We ourselves get confused by this: In `Bio.SeqIO.InsdcIO` we call a `CompoundLocation` a `CompoundFeatureLocation`, which does not exist.

We could change `FeatureLocation` to `SimpleLocation` in the biopython code base, and keep `FeatureLocation` as an obsolete alias to `SimpleLocation` for backward compatibility.

Reply:
Well spotted - there are a few uses of ``CompoundFeatureLocation`` in comments/docstrings which ought to be just ``CompoundLocation``, looks like when I wrote that the new class name was not yet fixed in my mind. I think shorter was better.

I agree with the compromise of keeping ``FeatureLocation`` as a backward compatible alias.